src/lib/utils/settings_manager.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <functional>
+#include <unordered_map>
+
+#include "types.hpp"
+#include "utils/settings/abstract_setting.hpp"
+
+namespace opossum {
+
+/**
+ * This component is the central point for accessing all Setting objects.
+ * Settings register here, so any component can get or change their values.
```

Comment:
You really like 'component', right? To me, it means only little more than 'thing'. Also, you are using the word twice here, referring to two different things.

"The SettingsManager is ..." and "Settings registered here can be changed from everywhere. Users change settings through the settings meta table."?